core/src/peer_id.rs

```diff
@@ -60,10 +60,9 @@ impl cmp::PartialOrd for PeerId {
 
 impl cmp::Ord for PeerId {
     fn cmp(&self, other: &Self) -> cmp::Ordering {
-        // must use borrow, because as_bytes is not consistent with equality
-        let lhs: &[u8] = self.borrow();
-        let rhs: &[u8] = other.borrow();
-        lhs.cmp(rhs)
+        let lhs = self.to_bytes();
```

Comment:
Given that `Multihash::to_bytes` allocates a `Vec` encoding the `Multihash` into it, would it not be quite expensive to sort a long list of `PeerId`s?

Reply:
Would it be possible to just use the digest (without the multihash code and the length) instead? Then we wouldn't need allocations.